server/modules/selva/module/aggregate.c

```diff
@@ -0,0 +1,1120 @@
+    struct SelvaObject *fields_obj = args->find_args.fields;
+    SVector *fields;
+    SelvaObject_GetArrayStr(fields_obj, "0", 1, NULL, &fields);
+
+    if (!fields) {
+        return 0;
+    }
+
+    struct SVectorIterator it;
+    SVector_ForeachBegin(&it, fields);
+
+    RedisModuleString *field;
+    while ((field = SVector_Foreach(&it))) {
+        enum SelvaObjectType field_type = SelvaObject_GetType(obj, field);
+        if (field_type == SELVA_OBJECT_LONGLONG) {
+            long long lv = 0;
+            SelvaObject_GetLongLong(obj, field, &lv);
+            if (lv < args->aggregation_result_double) {
+                args->aggregation_result_double = lv;
+            }
+
+            break;
+        } else if (field_type == SELVA_OBJECT_DOUBLE) {
+            double dv = 0;
+            SelvaObject_GetDouble(obj, field, &dv);
+            if (dv < args->aggregation_result_double) {
+                args->aggregation_result_double = dv;
+            }
+
+            break;
+        }
+    }
+
+    return 0;
+}
+
+static int agg_fn_max_obj(struct SelvaObject *obj, struct AggregateCommand_Args* args) {
+    struct SelvaObject *fields_obj = args->find_args.fields;
+    SVector *fields;
+    SelvaObject_GetArrayStr(fields_obj, "0", 1, NULL, &fields);
+
+    if (!fields) {
+        return 0;
+    }
+
+    struct SVectorIterator it;
+    SVector_ForeachBegin(&it, fields);
+
+    RedisModuleString *field;
+    while ((field = SVector_Foreach(&it))) {
+        enum SelvaObjectType field_type = SelvaObject_GetType(obj, field);
+        if (field_type == SELVA_OBJECT_LONGLONG) {
+            long long lv = 0;
+            SelvaObject_GetLongLong(obj, field, &lv);
+            if (lv > args->aggregation_result_double) {
+                args->aggregation_result_double = lv;
+            }
+
+            break;
+        } else if (field_type == SELVA_OBJECT_DOUBLE) {
+            double dv = 0;
+            SelvaObject_GetDouble(obj, field, &dv);
+            if (dv > args->aggregation_result_double) {
+                args->aggregation_result_double = dv;
+            }
+
+            break;
+        }
+    }
+
+    return 0;
+}
+
+
+static int apply_agg_fn_obj(struct SelvaObject *obj, struct AggregateCommand_Args* args) {
+    int err = 0;
+    if (args->aggregate_type == SELVA_AGGREGATE_TYPE_COUNT_NODE) {
+        err = agg_fn_count_obj(obj, args);
+    } else if (args->aggregate_type == SELVA_AGGREGATE_TYPE_COUNT_UNIQUE_FIELD) {
+        err = agg_fn_count_uniq_obj(obj, args);
+    } else if (args->aggregate_type == SELVA_AGGREGATE_TYPE_SUM_FIELD) {
+        err = agg_fn_sum_obj(obj, args);
+    } else if (args->aggregate_type == SELVA_AGGREGATE_TYPE_AVG_FIELD) {
+        err = agg_fn_avg_obj(obj, args);
+    } else if (args->aggregate_type == SELVA_AGGREGATE_TYPE_MIN_FIELD) {
+        err = agg_fn_min_obj(obj, args);
+    } else if (args->aggregate_type == SELVA_AGGREGATE_TYPE_MAX_FIELD) {
+        err = agg_fn_max_obj(obj, args);
+    }
```

Comment:
Assuming aggregate_type is already validated:

```suggestion
    const static int (*fun)(struct SelvaObject *, struct AggregateCommand_Args *)[] = {
        [SELVA_AGGREGATE_TYPE_COUNT_NODE] = agg_fn_count_obj,
        [SELVA_AGGREGATE_TYPE_COUNT_UNIQUE_FIELD] = agg_fn_count_uniq_obj,
        [SELVA_AGGREGATE_TYPE_SUM_FIELD] = agg_fn_sum_obj,
        [SELVA_AGGREGATE_TYPE_MIN_FIELD] = agg_fn_min_obj,
        [SELVA_AGGREGATE_TYPE_MAX_FIELD] = agg_fn_max_obj,
    };
    
    return fun[aggregate_type](obj, args);
```

Reply:
It's not validated, so I kept the old if actually.